### PR TITLE
Created shared button component

### PR DESCRIPTION
**Pull request by kaushik3pg**
### This PR: Creates a Shared Button Component. 
*  This is targeted against improving the code quality
* A lot of code uses <button> component which is duplicate code, hence created a new shared Button component.
* To use the new shared Button component, the import statement is : import Button from "../components/Button"
* Use the new shared Button component wherever you need to create a <button> tag.
* New shared Button component uses particular set of props, which can be found at following file : ../components/Button.js
* Use which ever props you require and can leave the others out, program should be able to handle the left out props.
* The reason for creating the new shared Button component is to remove redundant code, improve readability and in general how to write and structure the React code. 

PS. If this PR gets approved, I have a lot of ideas on improving this wonderful project.

### Self-Review Checklist
*(put x in between the brackets to mark them completed [x])*
- [x] PR title is clear with proper spelling and grammar
- [x] PR description contains a bulleted list of changes contained in the PR
- [x] PR links to relevant issues, with keywords to auto-close any issues fully resolved upon merge
- [x] All automated checks passed
- [x] Any variables introduced are named clearly and explicitly

### CODE REVIEW

File: components/Card.js [hunk 1]
```diff
@@ -60,30 +64,22 @@ CardParagraph.propTypes = {
...
+          ariaLabel="card-action-btn"
+          btnClasses={clsx('btn btn-accent shadow-md w-full h-auto', {
             loading: isPending,
           })}
-          onClick={onClick}
-          type="button"
-        >
-          {children}
-        </button>
+          btnLabel={children}
```
Comment (LowLifeArcade) [inline components/Card.js:80]: You seem to have forgotten `onClick` for your component.

File: components/Card.js [hunk 2]
```diff
@@ -93,7 +89,6 @@ CardAction.propTypes = {
   children: PropTypes.node,
   isPending: PropTypes.bool,
   linkTo: PropTypes.string,
-  onClick: PropTypes.func,
```
Comment (LowLifeArcade) [inline components/Card.js:96]: Or maybe you didn't forget it, but intentionally took it out for some reason?

**Review by LowLifeArcade — CHANGES_REQUESTED** [review]
Hi @kaushik3pg. I really appreciate all of the work you've done. And though I agree components are important, this button doesn't seem to simplify or create reusable code. It seems to be added code, in fact. Maybe you can rework your code to be more reusable. Like say make a `CardButton` which has all of the classes and aria-label abstracted away. A general button with all of the attributes exposed seems to be a little redundant here.

File: components/Card.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+
+import Button from '../components/Button';
+
 import React from 'react';
 import clsx from 'clsx';
-import { useContextTheme } from 'components/ThemeContext';
 
+import PropTypes from 'prop-types';
+
+import Image from 'next/image';
```
Comment (Sahil-101) [inline components/Card.js:10]: Hey @kaushik3pg, Just a small suggestion why keep the imports separated by a blank line.

**Review by Sahil-101 — COMMENTED** [review]
Good Idea @kaushik3pg, I really get the point of making a component so to have all button code at one place but essentially it doesn't reduces much code as of now if you give a look.

**Comment by kaushik3pg** [comment]
Hi, thanks for all the feedback. The idea behind creating the separate functional component is to access the same wherever needed. It looks like the code is not reduced much, but idea is to not create a new node every time(ie the <button>) and reuse what is available(ie.<Button>) just like a function call. So, we are not creating new functions every time and just calling the same function whenever required. Essentially what I did was, not created a new node on DOM everytime I needed a button, but just referenced the already created button component. 
For imports, to pass the eslint checks, they needed to be separated.
I am open to feedbacks on how to improve it further.

**Comment by LowLifeArcade** [comment]
> Hi, thanks for all the feedback. The idea behind creating the separate functional component is to access the same wherever needed. It looks like the code is not reduced much, but idea is to not create a new node every time(ie the button tag) and reuse what is available(ie. Button component) just like a function call. So, we are not creating new functions every time and just calling the same function whenever required. Essentially what I did was, not created a new node on DOM everytime I needed a button, but just referenced the already created button component.

As far as I know, a functional component returns jsx, so it is in fact creating new nodes every time it's called. Unless you can show me otherwise.

**Comment by LowLifeArcade** [comment]
> For imports, to pass the eslint checks, they needed to be separated. 

eslint is telling you to put them in alphabetical order, that's why you're getting that error. Spaces aren't the correct fix.

> I am open to feedbacks on how to improve it further.

I'm willing to put a PR through for you if you can submit something that has more usability. As it stands now it's not doing enough. Also, I will have to close this if it becomes inactive, so keep that in mind. Try to do more with it and have fun!

**Comment by LowLifeArcade** [comment]
Hey @kaushik3pg. I'm going to close this soon due to inactivity. Let me know if you need more time to work on it.

**Comment by idiglove** [comment]
Hi Im new here, but I understand where the author is coming from, I usually do this in my projects, this shared Button can be more customized this way, i.e, you want to inject a css module className or a new attribute, you'll just add a new prop, now you can do it for all the shared Buttons instead of going thru all instances of `buttons` and implementing that

**Pull request closed without merging**